compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -11768,9 +11775,7 @@ private STNode parseFromClause(boolean isRhsExpr) {
         STNode typedBindingPattern = parseTypedBindingPattern(ParserRuleContext.FROM_CLAUSE);
         STNode inKeyword = parseInKeyword();
 
-        // allow-actions flag is always false, since there will not be any actions
-        // within the from-clause, due to the precedence.
-        STNode expression = parseExpression(OperatorPrecedence.QUERY, isRhsExpr, false);
+        STNode expression = parseExpression(OperatorPrecedence.QUERY, isRhsExpr, true);
```

Comment:
We need to propagate the `allowActions` from `parseTerminalExpression` to here. We should only allow actions for `action-or-expr`.

For example,
[module-init-var-decl](https://ballerina.io/spec/lang/master/#module-init-var-decl) cannot have actions.
```bal
var _ = from var i in start f1() select i; // Should have error at module level
```

Reply:
Let's also remove the newline at L11777 now it is obsolete.